internal/archiver/archiver.go

```diff
@@ -476,7 +494,26 @@ func (arch *Archiver) save(ctx context.Context, snPath, target string, previous
 		return futureNode{}, true, nil
 	}
 
-	meta, err := arch.FS.OpenFile(target, fs.O_NOFOLLOW, true)
+	openFlag := fs.O_NOFOLLOW
+	if arch.ReadSpecial {
+		fileMode, err := arch.FS.Lstat(target)
+		if err != nil {
+			debug.Log("open metadata for %v returned error: %v", target, err)
+			return filterError(filterNotExist(err))
+		}
+		if fileMode.Mode&os.ModeSymlink != 0 {
+			targetFileMode, err := arch.FS.Stat(target)
```

Comment:
Exposing `Stat()` is unnecessary, just use `meta, err := arch.FS.OpenFile(target, 0, true); meta.Stat()` instead.